### PR TITLE
cmd/geth: fix js unclean shutdown

### PR DESCRIPTION
**Pull request by alex347**
This PR is a fix for #16506

### CODE REVIEW

**Review by holiman — APPROVED** [review]
Thanks a lot for fixing this old issue, after having seen the fix, I _was_ finally able to repro the issue and then confirm that the fix fixes it!

File: cmd/geth/consolecmd.go [hunk 1]
```diff
@@ -218,12 +216,9 @@ func ephemeralConsole(ctx *cli.Context) error {
 			utils.Fatalf("Failed to execute %s: %v", file, err)
 		}
 	}
-	// Wait for pending callbacks, but stop for Ctrl-C.
-	abort := make(chan os.Signal, 1)
-	signal.Notify(abort, syscall.SIGINT, syscall.SIGTERM)
 
 	go func() {
-		<-abort
+		stack.Wait()
```
Comment (holiman) [inline cmd/geth/consolecmd.go:221]: Actually, I think the fix would be even better if, `os.Exit(0)` were replaced by `console.Stop(false)`

Comment (alex347) [reply cmd/geth/consolecmd.go:221]: done

**Comment by JaneYork** [comment]
so，if I use geth1.10.4，the 'ctrl+c' always shut down Geth cleanly,at anytime?

**Pull request merged**